Razor renames and moves

NTaylorMullen opened the pull request:
- Renamed `CodeGenerator`s to `ChunkGenerator`s.
- Updated location of `TestFiles` from `TestFiles/CodeGenerator/CS/{Output|Source}` => `TestFiles/CodeGenerator/{Output|Source}`.
- Removed `ChunkTree` test; it was a legacy test used to experiment with Razor rendering (not a real test).
- Removed `CSharpRazorCodeGenerator`; it's now replaced with `RazorCodeGenerator`. It was an empty class that did nothing.
- Updated `ChunkBlock` => `ParentChunk`. Also updated several patterns throughout the code base that referenced these blocks as blocks and not parents.
- Moved chunks and chunk generators into the `Chunks`/`Chunks.Generators` namespace/folder structure. Updated test project to reflect the same.
- Moved `CodeBuilder`s and `CodeVisitor`s to the `CodeGeneration`/`CodeGeneration.Visitors` namespace/folder structure. Updated test project to reflect the same.
- Moved several TagHelper assets outside of their own namespaces and into Razors more general structures; such as CodeGeneration and Chunks/Chunks.Generators.
- Renamed CodeBuilder related classes to CodeGenerator
- Now that what used to be CodeGenerators are now ChunkGenerators we can rename the CodeBuilder into its correct structure: a CodeGenerator.
- Moved the `TagHelperAttributeValueCodeRenderer` from the `TagHelper`s namespace into the `CodeGeneration` namespace.
- Went through several classes and remove and sorted usings.
- Updated test files to abide by the new naming convention of `Builders` => `CodeGenerators`.

#140

NTaylorMullen commented:
/cc @dougbu

NTaylorMullen commented on src/Microsoft.AspNet.Razor/Chunks/Generators/TemplateBlockChunkGenerator.cs:
```diff
@@ -0,0 +1,20 @@
+
+namespace Microsoft.AspNet.Razor.Chunks.Generators
+{
+    public class TemplateBlockChunkGenerator : ParentChunkGenerator
```
I debated changing the various code generators names from `Block` => `Parent` but in the end decided not to sense the more specific Generators were there to relate to their corresponding syntax tree nodes.

pranavkm commented on src/Microsoft.AspNet.Razor/Chunks/ChunkTree.cs:
```diff
@@ -3,11 +3,11 @@
+namespace Microsoft.AspNet.Razor.Chunks
 {
-    public class CodeTree
+    public class ChunkTree
```
Does this type provide anything besides acting as a level of indirection - i.e. could we have `ChunkTreeBuilder` return an `IReadOnlyList<Chunk>` rather than have this type?

NTaylorMullen replied:
I debated it (was thinking same thing) but in the end decided not to do it purely to reduce churn since I couldn't come up with a strong enough reason to remove it.

pranavkm replied:
:+1: perhaps another iteration if we have time.

pranavkm commented on src/Microsoft.AspNet.Razor/Chunks/ExpressionBlockChunk.cs:
```diff
@@ -1,9 +1,9 @@
+namespace Microsoft.AspNet.Razor.Chunks
 {
-    public class ExpressionBlockChunk : ChunkBlock
+    public class ExpressionBlockChunk : ParentChunk
```
`ContainerChunk` maybe?

NTaylorMullen replied:
Not a fan, what does that add vs. what it is today.

pranavkm commented on src/Microsoft.AspNet.Razor/CodeGeneration/CodeGeneratorContext.cs:
```diff
@@ -24,7 +26,7 @@ public CodeBuilderContext(CodeGeneratorContext generatorContext, ErrorSink error
         }
 
         // Internal for testing.
-        internal CodeBuilderContext(RazorEngineHost host,
```
Might want to sort the indentation here.

pranavkm commented on src/Microsoft.AspNet.Razor/CodeGeneration/CodeGeneratorContext.cs:
```diff
@@ -24,7 +26,7 @@ public CodeBuilderContext(CodeGeneratorContext generatorContext, ErrorSink error
         // Internal for testing.
-        internal CodeBuilderContext(RazorEngineHost host,
+        internal CodeGeneratorContext(RazorEngineHost host,
                                     string className,
```
Here

pranavkm commented on test/Microsoft.AspNet.Razor.Test/CodeGeneration/AttributeBlockChunkGeneratorTest.cs:
```diff
@@ -4,36 +4,36 @@
 using Xunit;
 
-namespace Microsoft.AspNet.Razor.Generator
+namespace Microsoft.AspNet.Razor.Chunks.Generators
```
Could you also move the type to `/test/Microsoft.AspNet.Razor.Test/Chunks/Generators` directory?

NTaylorMullen replied:
:+1:

pranavkm commented:
:shipit:

dougbu commented on src/Microsoft.AspNet.Razor/Chunks/ChunkTreeBuilder.cs:
```diff
@@ -4,20 +4,20 @@
+    public class ChunkTreeBuilder
     {
-        private readonly Stack<ChunkBlock> _blockChain;
+        private readonly Stack<ParentChunk> _parentChain;
```
`_parentStack` ?

dougbu commented on src/Microsoft.AspNet.Razor/CSharpRazorCodeLanguage.cs:
```diff
@@ -1,9 +1,8 @@
-using Microsoft.AspNet.Razor.Generator.Compiler;
-using Microsoft.AspNet.Razor.Generator.Compiler.CSharp;
+using Microsoft.AspNet.Razor.Chunks.Generators;
+using Microsoft.AspNet.Razor.CodeGeneration;
```
Namespaces look inconsistent when side-by-side. Why not "Generators" for both chunks and code?

NTaylorMullen replied:
Because there's not multiple code generators today :cry:

dougbu replied:
suggest `Microsoft.AspNet.Razor.CodeGenerator`

NTaylorMullen replied:
Will do `.CodeGenerators` so it doesn't conflict with the common Razor `CodeGenerator` type.

dougbu commented on src/Microsoft.AspNet.Razor/Chunks/Generators/ChunkGeneratorContext.cs:
```diff
@@ -0,0 +1,44 @@
+{
+    public class ChunkGeneratorContext
+    {
+        protected ChunkGeneratorContext(ChunkGeneratorContext context)
```
Is this file new or a rename of something else?

If GitHub is just confused, please add comments letting us know the previous file names (if possible).

NTaylorMullen replied:
`CodeGeneratorContext` was its old name.

dougbu commented on src/Microsoft.AspNet.Razor/Chunks/Generators/HybridChunkGenerator.cs:
```diff
@@ -0,0 +1,22 @@
+
+namespace Microsoft.AspNet.Razor.Chunks.Generators
+{
+    public abstract class HybridChunkGenerator : ISpanChunkGenerator, IParentChunkGenerator
```
Can we please get rid of this useless class?

NTaylorMullen replied:
Surreee why not :smile:

NTaylorMullen commented on src/Microsoft.AspNet.Razor/CodeGeneration/CodeGenerator.cs:
```diff
@@ -0,0 +1,22 @@
+
+namespace Microsoft.AspNet.Razor.CodeGeneration
+{
+    public abstract class CodeGenerator
```
Used to be called `CodeBuilder.cs`

NTaylorMullen commented on src/Microsoft.AspNet.Razor/Chunks/Generators/IParentChunkGenerator.cs:
```diff
@@ -0,0 +1,13 @@
+
+namespace Microsoft.AspNet.Razor.Chunks.Generators
+{
+    public interface IParentChunkGenerator
```
Used to be called `IBlockCodeGenerator.cs`

dougbu commented on src/Microsoft.AspNet.Razor/Chunks/Generators/ChunkGeneratorContext.cs:
```diff
@@ -0,0 +1,44 @@
+
+        public string SourceFile { get; internal set; }
+
+        public string RootNamespace { get; private set; }
```
Do we need the `private set`ters for anything?

dougbu commented on src/Microsoft.AspNet.Razor/CodeGeneration/Visitors/CSharpTagHelperAttributeValueVisitor.cs:
```diff
@@ -97,7 +98,7 @@ protected override void Visit(LiteralChunk chunk)
         /// <remarks>
         /// Allowed to support future C# extensions. Likely "~/..." will lead to a C# compilation error but that is up
-        /// to the compiler.
+        /// to the .
```
Need some word here if you're going to remove `/// ... compiler`

NTaylorMullen commented:
![image](https://cloud.githubusercontent.com/assets/2008729/7871570/9aa813be-0545-11e5-8097-7dc6b94dbc22.png)
![image](https://cloud.githubusercontent.com/assets/2008729/7871603/c96b3e88-0545-11e5-9682-a4d763531cd1.png)

dougbu commented on src/Microsoft.AspNet.Razor/GeneratorResults.cs:
```diff
@@ -2,7 +2,8 @@
 using System.Collections.Generic;
-using Microsoft.AspNet.Razor.Generator.Compiler;
+using Microsoft.AspNet.Razor.Chunks;
+using Microsoft.AspNet.Razor.CodeGeneration;
```
This class should be in the `Microsoft.AspNet.Razor.CodeGenerator` namespace. Wouldn't need the `using` then.

dougbu commented on src/Microsoft.AspNet.Razor/RazorResources.resx:
```diff
@@ -376,7 +376,7 @@ Instead, wrap the contents of the block in "{{}}":
   </data>
   <data name="TagHelpers_TagHelperCodeGeneartorMustBeAssociatedWithATagHelperBlock" xml:space="preserve">
-    <value>A TagHelperCodeGenerator must only be used with TagHelperBlocks.</value>
+    <value>A TagHelperChunkGenerator must only be used with TagHelperBlocks.</value>
```
Hmm, the names `TagHelperChunkGenerator` and `TagHelperBlock` should be passed into this resource. Not that the message is all that useful externally -- sounds like a `Debug.Assert()` message.

dougbu commented:
Please take a pass at the long lines that are getting longer (the excessive ones -- 150 chars or so :smiley_cat:).

dougbu commented:
:watch: just for the larger rename or two I recommended.

(Thanks for the pic. Looks like the namespaces / folders are all of a reasonable size.)

NTaylorMullen commented:
Addressed code review comments. Started going through the long line fixes then realized it was taking up too much time (they were already long). Razor is a disaster zone when it comes to long lines :cry:

dougbu commented on src/Microsoft.AspNet.Razor/Chunks/ChunkTreeBuilder.cs:
```diff
@@ -27,13 +27,13 @@ public void AddChunk(Chunk chunk, SyntaxTreeNode association, bool topLevel = fa
             chunk.Association = association;
 
             // If we're not in the middle of a chunk block
```
"parent chunk"

dougbu replied:
This was the only comment issue I caught but it may be worth doing global searches for "chunk block" and "block code".

dougbu commented on src/Microsoft.AspNet.Razor/Chunks/Generators/ExpressionChunkGenerator.cs:
```diff
@@ -1,28 +1,27 @@
+namespace Microsoft.AspNet.Razor.Chunks.Generators
 {
-    public class ExpressionCodeGenerator : HybridCodeGenerator
+    public class ExpressionChunkGenerator : ISpanChunkGenerator, IParentChunkGenerator
```
thx :clap:

dougbu commented:
> Razor is a disaster zone when it comes to long lines :cry:

Agreed and you improved things more than I'd hoped. You even started on wrapping long lines in test code.

dougbu commented:
:shipit: once reactions are done. Any other repos besides Diagnostics and MVC affected?
